openedx/core/djangoapps/user_authn/exceptions.py

```diff
@@ -10,18 +10,19 @@ class AuthFailedError(Exception):
     message.
     """
     def __init__(
-        self, value=None, redirect=None, redirect_url=None, error_code=None
+        self, value=None, redirect=None, redirect_url=None, error_code=None, context={},
     ):
         super(AuthFailedError, self).__init__()
         self.value = Text(value)
         self.redirect = redirect
         self.redirect_url = redirect_url
         self.error_code = error_code
+        self.context = context
 
     def get_response(self):
         """ Returns a dict representation of the error. """
         resp = {'success': False}
-        for attr in ('value', 'redirect', 'redirect_url', 'error_code'):
+        for attr in ('value', 'redirect', 'redirect_url', 'error_code', 'context'):
```

Comment:
@waheedahmed the reason behind adding this it to avoid adding a lot of logic under here: 
https://github.com/edx/edx-platform/blob/26de4961ab0c247294f63c1931d2e595d96282c1/openedx/core/djangoapps/user_authn/views/login.py#L509-L510

Also this will help us in our move to use `error_code` on MFE instead of sending error message to MFE. Context will help us to send any related data to MFE specific to that error code

Reply:
Make sense 👍